projects/packages/publicize/src/class-publicize-script-data.php

```diff
@@ -267,7 +267,21 @@ public static function has_feature_flag( $feature ): bool {
 			return true;
 		}
 
-		return Current_Plan::supports( 'social-' . $feature );
+		$feature_name = 'social-' . $feature;
+
+		if ( Current_Plan::supports( $feature_name ) ) {
+			return true;
+		}
+
+		// Currently, Current_Plan::supports() doesn't work well for Simple sites.
+		// So, we need to do some extra work to check for Simple sites.
+		if ( ! ( new Host() )->is_wpcom_simple() ) {
+			return false;
+		}
+
+		$features = Current_Plan::get_simple_site_specific_features();
+
+		return ! empty( $features['active'] ) && in_array( $feature_name, $features['active'], true );
```

Comment:
Could we put this logic in Current_Plan::supports? Or just in Current_Plan when it gets the data?

Otherwise we'll keep having to do this as whenever we use Current_Plan::supports

Reply:
Ideally, we should update [WPCOM_Features::has_feature()](https://github.com/Automattic/jetpack/blob/65c48ba33af6107b535943c8d75d3bd93c41662a/projects/plugins/wpcomsh/wpcom-features/class-wpcom-features.php#L1525) to also check for the features added via `Store_Product_List::get_site_specific_features_data()` on WPCOM.

But, I am not sure if that will have any unexpected results for existing usage of `Current_Plan::supports()`

Reply:
Please see 9d2b471292947f64e43c5b9dd8c410ca0c5a4353

Reply:
That change makes the WPCOM plan-related tests fail.